### PR TITLE
Minigame Snake Game Complete

### PR DESCRIPTION
**Pull request by EllaBerglas**
Team 7 pull request for sprint 1. Fully finished snake mini game.

### CODE REVIEW

**Review by TarushiG — COMMENTED** [review]
Looks good! The snake game's working great on my PC.

**Review by TarushiG — APPROVED** [review]

**Review by shaivikaaaa — APPROVED** [review]
looks good to me!

File: source/core/src/main/com/csse3200/game/components/minigame/MiniGameConstants.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package com.csse3200.game.components.minigame;
```
Comment (BaileyS03) [inline source/core/src/main/com/csse3200/game/components/minigame/MiniGameConstants.java:1]: No tests here

File: source/core/src/main/com/csse3200/game/components/minigame/MiniGamesScores.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
+package com.csse3200.game.components.minigame;
+
```
Comment (BaileyS03) [inline source/core/src/main/com/csse3200/game/components/minigame/MiniGamesScores.java:2]: 64% coverage

File: source/core/src/main/com/csse3200/game/components/minigame/snake/rendering/AppleRenderer.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
+package com.csse3200.game.components.minigame.snake.rendering;
+
+import com.badlogic.gdx.Gdx;
```
Comment (BaileyS03) [inline source/core/src/main/com/csse3200/game/components/minigame/snake/rendering/AppleRenderer.java:3]: No unit tests but visually tests alright

File: source/core/src/main/com/csse3200/game/components/minigame/snake/rendering/GridRenderer.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
+package com.csse3200.game.components.minigame.snake.rendering;
+
+
```
Comment (BaileyS03) [inline source/core/src/main/com/csse3200/game/components/minigame/snake/rendering/GridRenderer.java:3]: No unit tests but game would brick if it didn't work so visually tests okay

File: source/core/src/main/com/csse3200/game/components/minigame/snake/rendering/SnakeGameRenderer.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+import com.csse3200.game.services.ServiceLocator;
+import com.csse3200.game.ui.minigame.SnakeScoreBoard;
+import static com.csse3200.game.components.minigame.snake.AssetPaths.IMAGES;
+
+/**
+ * Renders all elements of the Snake mini-game, including the grid, apple, snake, and scoreboard.
+ */
+public class SnakeGameRenderer {
+
+    private final GridRenderer gridRenderer;
```
Comment (BaileyS03) [inline source/core/src/main/com/csse3200/game/components/minigame/snake/rendering/SnakeGameRenderer.java:18]: Same

**Review by BaileyS03 — APPROVED** [review]

File: source/core/src/main/com/csse3200/game/components/minigame/MiniGamesScores.java (hunk 2 above)
Comment (EllaBerglas) [reply source/core/src/main/com/csse3200/game/components/minigame/MiniGamesScores.java:2]: Yep, will implement more tests as the code is completed for that with all 3 minigames.

**Pull request merged**